### PR TITLE
Add scripts to automate the AML and TAA registration and update process

### PR DESCRIPTION
**Pull request by WadeBarnes**
Signed-off-by: Wade Barnes <wade@neoterictech.ca>

### CODE REVIEW

File: manage [hunk 1]
```diff
@@ -101,6 +101,43 @@ usage () {
...
+              amlVersion
+                - The version of the AML.
+              taaUrl
+                - The URL to the raw content of the TAA in markdown format.
+              taaVersion
+                - The vesion of the TAA
+              taaRatificationTime
+                - The date and time of TAA ratification by network government.
+
+        Example:
```
Comment (ianco) [inline manage:130]: Should this example work?  When I try (against a local von-network) I get:

```
$ ./manage apply-taa                 walletName=local_net_trustee_wallet                 poolName=local_net                 useDid=V4SGRU86Z58d6TV7PBUe6f                 amlUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Acceptance-Mechanism-List-(AML).md'                 amlVersion=0.1                 taaUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Transaction-Author-Agreement.md'                 taaVersion=0.1                 taaRatificationTime="2022-08-09T11:05:00-0700"

Using: docker-compose --log-level ERROR

date: illegal option -- -
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ... 
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
sed: 1: "/{/,/}/{/{/!{/}/!p}}": extra characters at the end of p command
Creating von_client_run ... done
Can't read the file
Goodbye...
```

Comment (WadeBarnes) [reply manage:130]: Yes, it should work.  That is the command I was using for testing.

Comment (WadeBarnes) [reply manage:130]: Do you get the same error if you run this command:
```
date --date="2022-08-09T11:05:00-0700" +"%s"
```

**Comment by swcurran** [comment]
With that command, I get an error on a Mac, works on Linux.

**Comment by WadeBarnes** [comment]
The syntax on Mac is different.  it's will be something like:
```
date -jf "%Y-%m-%dT%H:%M:%S%Z" "2022-08-09T11:05:00-0700" +%s
```

Would one of you be able to confirm?

File: manage (hunk 1 above)
Comment (ianco) [reply manage:130]: > Do you get the same error if you run this command?
> 
> ```
> date --date="2022-08-09T11:05:00-0700" +"%s"
> ```
> 
> The expected result is `1660068300`

Yes same error, I'll try your mac version

**Comment by ianco** [comment]
> date -jf "%Y-%m-%dT%H:%M:%S%Z" "2022-08-09T11:05:00-0700" +%s

```

**Comment by ianco** [comment]
This works, I think it automatically pucks up the local timezone:

`date -j -f "%Y-%m-%dT%H:%M:%S" "2022-08-09T11:05:00" +%s`

**Comment by WadeBarnes** [comment]
It would be nice to be able to specify the actual time offset since the ratification time and the person running the script may not be in the same time zone.

**Comment by WadeBarnes** [comment]
Looks like `Z%` for the MAC command needs to be an alphabetic time zone abbreviation.

So something like this should work:
`date -jf "%Y-%m-%dT%H:%M:%S %Z" "2022-08-09T11:05:00 PDT" +%s`

Expected result is `1660068300`

@ianco, would you be able to try this out when you have a moment?

**Comment by swcurran** [comment]
Works for Mac, not for Linux/gitBash (as expected...).

File: manage (hunk 1 above)
Comment (WadeBarnes) [reply manage:139]: Should be fixed in the most recent commit.

**Comment by WadeBarnes** [comment]
Turns out Linux (and Git Bash) can use the "%Y-%m-%dT%H:%M:%S %Z" format without complaint too.  So I've updated the script to support Mac and updated the example and docs to use the the "%Y-%m-%dT%H:%M:%S %Z" date format.

File: manage [hunk 2]
```diff
@@ -516,6 +562,65 @@ function generateGenesisFiles() {
...
+  # Create temp files for the AML and ATT
+  amlPath="./tmp/aml-$(date +"%s").txt"
+  taaPath="./tmp/att-$(date +"%s").txt"
+
+  # Download the AML and ATT files
+  curl -s -o ${amlPath} "${amlUrl}"
+  curl -s -o ${taaPath} "${taaUrl}"
+
+  # Parse the AML content from the file;the bit inside the '{}'s
+  amlContent="{$(sed -n '/{/,/}/{/{/!{/}/!p}}' ${amlPath} | sed 's/^ *//g')}"
```
[the remark this answers is not in the document]
Comment (ianco) [reply manage:607]: This is the Mac variant:  `sed -n '/{/,/}/{/{/!{/}/!p;};}' ${amlPath}`

**Review by ianco — CHANGES_REQUESTED** [review]

File: manage (hunk 2 above)
[the remark this answers is not in the document]
Comment (WadeBarnes) [reply manage]: That works in git bash too.  Code updated.

File: manage [hunk 3]
```diff
@@ -101,6 +101,52 @@ usage () {
...
+                - The date and time of TAA ratification by network government.
+                - On Linux and Windows (Git Bash):
+                  - The date format is "%Y-%m-%dT%H:%M:%S%Z", where `%Z` is a numeric time zone offset,
+                    or "%Y-%m-%dT%H:%M:%S %Z", where `%Z` is an alphabetic time zone abbreviation.
+                  - Examples: 
+                    - "2022-08-09T11:05:00-0700"
+                    - "2022-08-09T11:05:00 PDT"
+                - On Mac, the date format is "%Y-%m-%dT%H:%M:%S %Z", where `%Z` is an alphabetic time zone abbreviation.
+                  - Example:
+                    - "2022-08-09T11:05:00 PDT"
```
[the remark this answers is not in the document]
Comment (WadeBarnes) [reply manage]: Code updated to reflect this.

**Comment by WadeBarnes** [comment]
Code updated based on @ianco's review.

**Review by ianco — APPROVED** [review]
Added local file ./tmp/cliconfig.json

./manage cli init-pool local_net http://host.docker.internal:9000/genesis

./manage indy-cli create-wallet walletName=local_net_trustee_wallet

./manage apply-taa                 walletName=local_net_trustee_wallet                 poolName=local_net                 useDid=V4SGRU86Z58d6TV7PBUe6f                 amlUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Acceptance-Mechanism-List-(AML).md'                 amlVersion=0.1                 taaUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Transaction-Author-Agreement.md'                 taaVersion=0.1                 taaRatificationTime="2022-08-09T11:05:00PDT"

**Pull request merged**